pkg/util/flag/flags.go

```diff
@@ -12,18 +12,119 @@ distributed under the License is distributed on an "AS IS" BASIS,
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
-
-@CHANGELOG
-KubeEdge Authors: delete useless functions
 */
 
 package flag
 
 import (
+	"fmt"
+	"os"
+	"strconv"
+
 	"github.com/spf13/pflag"
 	"k8s.io/klog"
+	"sigs.k8s.io/yaml"
+
+	"github.com/kubeedge/kubeedge/pkg/apis/cloudcore/v1alpha1"
+)
+
+type configValue int
+
+const (
+	ConfigFalse configValue = 0
+	ConfigTrue  configValue = 1
 )
 
+func (m *configValue) IsBoolFlag() bool {
+	return true
+}
+
+func (m *configValue) Get() interface{} {
+	return configValue(*m)
+}
+
+func (m *configValue) Set(s string) error {
+	boolVal, err := strconv.ParseBool(s)
+	if boolVal {
+		*m = ConfigTrue
+	} else {
+		*m = ConfigFalse
+	}
+	return err
+}
+
+func (m *configValue) String() string {
+	return fmt.Sprintf("%v", bool(*m == ConfigTrue))
+}
+
+// The type of the flag as required by the pflag.Value interface
+func (m *configValue) Type() string {
+	return "config"
+}
+
+func ConfigVar(p *configValue, name string, value configValue, usage string) {
+	*p = value
+	pflag.Var(p, name, usage)
+	pflag.Lookup(name).NoOptDefVal = "true"
+}
+
+func Config(name string, value configValue, usage string) *configValue {
+	p := new(configValue)
+	ConfigVar(p, name, value, usage)
+	return p
+}
+
+const minConfigFlagName = "minconfig"
+const defaultConfigFlagName = "defaultconfig"
+
+var (
+	minConfigFlag     = Config(minConfigFlagName, ConfigFalse, "Print min configuration for reference, users can refer to it to create their own configuration files, it is suitable for beginners.")
+	defaultConfigFlag = Config(defaultConfigFlagName, ConfigFalse, "Print default configuration for reference, users can refer to it to create their own configuration files, it is suitable for advanced users.")
+)
+
+// AddFlags registers this package's flags on arbitrary FlagSets, such that they point to the
+// same value as the global flags.
+func AddFlags(fs *pflag.FlagSet) {
+	fs.AddFlag(pflag.Lookup(minConfigFlagName))
+	fs.AddFlag(pflag.Lookup(defaultConfigFlagName))
+}
+
+// PrintMinConfigAndExitIfRequested will check if the -minconfig flag was passed
+// and, if so, print the min config and exit.
+func PrintMinConfigAndExitIfRequested() {
+	if *minConfigFlag == ConfigTrue {
+		config := v1alpha1.NewMinCloudCoreConfig()
```

Comment:
How about set the `v1alpha1.NewMinCloudCoreConfig()` as a parameter for PrintMinConfigAndExitIfRequested？

Reply:
sound good